include/glow/ExecutionEngine/ExecutionEngine.h

```diff
@@ -67,11 +67,17 @@ class ExecutionEngine final {
   /// \returns the compiled function.
   CompiledFunction &getCompiledFunction() { return *function_; }
 
-  /// \returns whether operation is supported by the underlying backend.
-  bool isOpSupported(Kinded::Kind opKind, ElemKind elementTy) const {
-    return backend_->isOpSupported(opKind, elementTy);
+  /// \returns a cost associated with executing the provided \p opKind with \p
+  /// inputTypes and \p outputTypes on the underlying backend.
+  CostVal getOpCost(Kinded::Kind opKind, llvm::ArrayRef<TypeRef> inputTypes,
+                    llvm::ArrayRef<TypeRef> outputTypes) const {
+    return backend_->getOpCost(opKind, inputTypes, outputTypes);
   }
 
+  /// \returns whether the cost associated with executing Node \p N on the
+  /// underlying backend.
+  CostVal getOpCost(const Node &N) const { return backend_->getOpCost(N); }
```

Comment:
I think maybe this function need to be `virtual` for backend to provide more accurate estimation.
because the cost is depend on operator's parameter., 
For example, if the input size is too small, the CPU maybe faster than the ASIC.

Reply:
The actual logic for computing an op's cost is contained in a virtual `getOpCost()`:

https://github.com/pytorch/glow/blob/0c01a667b8fb26f7752ae7ed63c4b6d134ed729c/include/glow/Backends/Backend.h#L78-L82

This function here in ExecutionEngine just a helper method. It forwards to the backend's helper `getOpCost(Node &N)`, which then calls into each backend's implementation of `getOpCost()`.

Reply:
Thanks, I was missed the `TypeRef` can represents a type of tensor. 
Sorry to bother you.

Reply:
No bother at all, thanks for checking out the PR! We are happy to hear any questions or insights from everyone contributing to the project.